client/src/lib.rs

```diff
@@ -679,7 +679,7 @@ fn parse_logs_response<T: anchor_lang::Event + anchor_lang::AnchorDeserialize>(
         if let Ok(mut execution) = Execution::new(&mut logs) {
             // Create a new peekable iterator so that we can peek at the next log whilst iterating
             let mut logs_iter = logs.iter().peekable();
-            let regex = Regex::new(r"^Program (.*) invoke.*$").unwrap();
+            let regex = Regex::new(r"^Program ([1-9A-HJ-NP-Za-km-z]+) invoke \[[\d]+\]$").unwrap();
```

Comment:
using the `+` will allow unbounded string lengths, even though Solana base58 addresses must be between 32-44 characters long (inclusive)

```suggestion
            let regex = Regex::new(r"^Program ([1-9A-HJ-NP-Za-km-z]{32,44}) invoke \[[\d]+\]$").unwrap();
```

Reply:
https://github.com/anza-xyz/agave/blob/5fa721b3b27c7ba33e5b0e1c55326241bb403bb1/program-runtime/src/stable_log.rs

the two spaces between the regex already prevented that